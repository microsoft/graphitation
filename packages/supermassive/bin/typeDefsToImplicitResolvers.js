#!/usr/bin/env node

(function () {
  require("@graphitation/supermassive/lib/bin/typeDefsToImplicitResolvers")
    .typeDefsToImplicitResolvers()
    .parseAsync(process.argv);
})();
