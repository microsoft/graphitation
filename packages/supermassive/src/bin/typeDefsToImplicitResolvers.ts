export async function typeDefsToImplicitResolvers(filename: string): void {}
