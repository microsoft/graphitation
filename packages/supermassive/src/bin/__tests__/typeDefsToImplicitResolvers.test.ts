import fs from "fs/promises";
import path from "path";
import { typeDefsToImplicitResolvers } from "../typeDefsToImplicitResolvers";

describe(typeDefsToImplicitResolvers, () => {
  beforeAll(async () => {
    await fs.rm(path.join(__dirname, "fixtures", "__generated__"), {
      recursive: true,
      force: true,
    });
  });

  afterAll(async () => {
    await fs.rm(path.join(__dirname, "fixtures", "__generated__"), {
      recursive: true,
      force: true,
    });
  });

  it("should generate", async () => {
    const program = typeDefsToImplicitResolvers();
    await program.parseAsync(
      [path.join(__dirname, "./fixtures/schema.graphql")],
      { from: "user" }
    );
    expect(
      await fs.readFile(
        path.join(__dirname, "./fixtures/__generated__/schema.ts"),
        { encoding: "utf-8" }
      )
    ).toMatchSnapshot();
  });
});
