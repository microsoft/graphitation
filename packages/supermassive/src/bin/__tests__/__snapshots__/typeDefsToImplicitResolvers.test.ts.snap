// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supermassive should generate 1`] = `
"import { GraphQLList, GraphQLNonNull, GraphQLID, GraphQLString, GraphQLInt, GraphQLFloat, GraphQLBoolean, GraphQLEnumType } from \\"graphql\\";
const SearchResult = { __types: [\\"Person\\", \\"Starship\\", \\"Transport\\", \\"Species\\", \\"Vehicle\\", \\"Planet\\", \\"Film\\"], __resolveType: undefined };
const NodeType = new GraphQLEnumType({ name: \\"NodeType\\", description: \\"\\", values: { Person: { description: \\"\\" }, Starship: { description: \\"\\" }, Transport: { description: \\"\\" }, Species: { description: \\"\\" }, Vehicle: { description: \\"\\" }, Planet: { description: \\"\\" }, Film: { description: \\"\\" } } });
const Query = {};
const Film = {};
const Vehicle = {};
const Person = {};
const Starship = {};
const Planet = {};
const Species = {};
const Transport = {};
const Node = { __implementedBy: [\\"Film\\", \\"Vehicle\\", \\"Person\\", \\"Starship\\", \\"Planet\\", \\"Species\\", \\"Transport\\"], __resolveType: undefined };
export const resolvers = { SearchResult, NodeType, Query, Node, Film, Vehicle, Person, Starship, Planet, Species, Transport };
"
`;
